### DISCUSSION TITLE
Multiply Instead of Repeats - Option to Train on a Portion of Images Per Epoch

### DISCUSSION THREAD

**Discussion by a-l-e-x-d-s-9** [kohya-ss/sd-scripts#819] [Ideas]
I'm familiar with the [EveryDread2](https://github.com/victorchall/EveryDream2trainer) (ED2) trainer, which provides an option to specify any portion of images from a folder. Instead of simply defining the number of repeats for a folder, you can set a 0.5 multiplier for a folder, and it will train on half of the images, selected randomly each epoch. You can also use a 1.6 multiplier or any other positive number. Negative number will discard the folder, which also important.
This approach not only includes the concept of repeats but also offers greater flexibility. It's a more effective way to balance between subjects without the need to discard any images.
I've utilized this feature in every training session with ED2, and I'm convinced that it's a crucial feature for the Kohya trainer as well.
If anyone is familiar with the Kohya codebase, could you please assess whether it's feasible to implement this feature in a reasonable manner? Your feedback would be appreciated.

**Comment by wireframewares** [comment]
Coming from EveryDream2 as well as OneTrainer, it's absolutely backwards that Kohya doesn't support this.

Being able to set a <1 decimal value in ED2 and OneTrainer is so practical and flexible, **_such as random image selection per epoch_**, as well as the sheer convenience of mixing datasets on the fly (training a person with 50 images, and mixing it with a folder full of 2,000 images of regularization ---> simply set the regularization folder to Repeats = 0.025 ... rather than having to manually remake the folder into 50 images!).
